### ISSUE TITLE
train_test_split() does not preserve correct lengths across X and y

### ISSUE THREAD

**Issue by py-rh** [dask/dask-ml#1005]
```
ddf = dd.read_parquet('gs://demo/bitcoin-2024/') # Toy example, ~150GB 

features = ['is_coinbase', 'input_count','output_count','virtual_size','version']
target_var = 'size'

X = ddf[features] 
y = ddf[target_var]

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.3, shuffle=False
)

print(f"Total rows in X: {len(X)}")
print(f"Total rows in y: {len(y)}")
# Total rows in X: 184940856
# Total rows in y: 184940856

print(f"Total rows in X_train: {len(X_train)}")
print(f"Total rows in y_train: {len(y_train)}")
# Total rows in X_train: 129453585
# Total rows in y_train: 129456685

```

Please let me know if I am doing something wrong

**Comment by py-rh** [comment]
X_train, y_train = X_train.align(y_train, join='inner', axis=0)

**Comment by navashiva** [comment]
@py-rh I understand that the recommendation was a workaround. I wonder if we have identified the root cause of the behavior resulting in different rows for X_train and y_train.
